addons/crm/static/src/js/forecast/forecast_controllers.js

```diff
@@ -0,0 +1,39 @@
+/** @odoo-module alias=crm.ForecastControllers **/
+import KanbanController from 'web.KanbanController';
+
+var ForecastKanbanController = KanbanController.extend({
+    custom_events: _.extend({}, KanbanController.prototype.custom_events, {
+        forecast_kanban_add_column: '_onAddColumnForecast',
+    }),
+    /**
+     * Expand the fill_temporal period after the ForecastColumnQuickCreate has been used,
+     * then reload the view to refetch updated data
+     * @param {OdooEvent} ev 
+     */
+    _onAddColumnForecast(ev) {
+        ev.stopPropagation();
+        this.call('fillTemporalService', 'getFillTemporalPeriod', {
+            modelName: this.model.loadParams.modelName,
+            field: {
+                name: this.model.forecast_field,
+                type: this.model.loadParams.fields[this.model.forecast_field].type,
+            },
+            granularity: this.model.granularity,
+        }).expand();
+        const self = this;
+        this.mutex.exec(function () {
+            return self.update(
+                {groupBy:[`${self.model.forecast_field}:${self.model.granularity}`],
+                 forceFillingTo: true
+                },
+                {reload: true}
+            );
+        }).then(function () {
+            return self.renderer.trigger_up("quick_create_column_created");
+        });
```

Comment:
Arrow functions are great in these cases since it allows you to use `this` in nested functions:
```suggestion
        this.mutex.exec(() => this.update(
            {
                groupBy: [`${this.model.forecast_field}:${this.model.granularity}`],
                forceFillingTo: true
            },
            { reload: true }
        );
```
... and I have mixed feelings about this one, know what is it needed for?:
```js
        return this.renderer.trigger_up("quick_create_column_created");
```

Reply:
It's an event for the mobile kanban view but since I did not implement the add column for mobile yet, it is not yet useful